llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -6908,7 +6908,8 @@ static SDValue combineSelectToBinOp(SDNode *N, SelectionDAG &DAG,
   MVT VT = N->getSimpleValueType(0);
   SDLoc DL(N);
 
-  if (!Subtarget.hasShortForwardBranchOpt()) {
+  if (!Subtarget.hasShortForwardBranchOpt() &&
```

Comment:
The pattern is a subset of ShortForwardBranch so yes. Would need to make ShortForwardBranch imply TuneCMOVBranchOpt in tablegen which I think is doable.

Reply:
Actually its more complicated because ShortForwardBranch can fuse `mv` or `c.mv` but CMOVBranchOpt requires C. So I don't think the dependency alone would be enough, but we could have a single subtarget function to wrap this.